### PR TITLE
Envpool's Async PPO

### PR DESCRIPTION
**Pull request by vwxyzjn**
This PR prototypes a PPO that works with Envpool's Async API, which produces `ppo_atari_envpool_async_jax.py`.  I tried to do everything I could to ensure the implementation is okay, but its losses (black lines) look slightly off than the sync PPO baseline (red lines) below: notice the entropy curve especially looked much higher.

Screen Shot 2022-07-28 at 9 39 37 AM

To further investigate this, I tried to see if envpool's Async API is the issue. To achieve this, I created two `FakeAsyncEnvs`  with the same API as envpool's async API interface. The first `FakeAsyncEnvs` uses the envpool's synchronous under the hood (`ppo_atari_envpool_fake_async_jax.py`), and the second uses gym's atari env under the hood (`ppo_atari_gym_fake_async_jax.py`).  And they result in the same loss curves (purple and dark red curves) as the synchronous version. 

@mavenlin @Trinkle23897 could you look at if my `FakeAsyncEnvs` are correct? if they are, it probably means there is a bug with envpool's async API...

https://github.com/vwxyzjn/envpool-xla-cleanrl/blob/f1db556c5708f66f6ec943fad606538eca800836/ppo_atari_envpool_fake_async_jax.py#L92-L147

https://github.com/vwxyzjn/envpool-xla-cleanrl/blob/f1db556c5708f66f6ec943fad606538eca800836/ppo_atari_gym_fake_async_jax.py#L123-L175

### CODE REVIEW

**Comment by vwxyzjn** [comment]
Turns out the bug is that I made the assumption that in envpool's async API different batches would have distinct ids. For example, if num_envs = 8, batch_size = 4, I assume the env_ids in two batches would look like

```
[0,2,7,4], [1, 3, 6, 5] # distinct IDs in different batches
```

but in reality, envpool's async API will return whichever returns the fastest, so it's possible to have duplicate IDs in different batches, like

```
[0,2,7,4], [1, 0, 6, 5] # some duplicate IDs in different batches
```

This means doing 1 step in env 3 is more slower than doing 2 steps in env 0, which could happen.

I built a `ppo_atari_envpool_soft_async_jax.py` to ensure duplicate IDs do not happen.
